src/input/mod.rs

```diff
@@ -2221,6 +2247,18 @@ fn should_intercept_key(
         }
     }
 
+    if is_inhibiting_shortcuts
```

Comment:
I have a feeling that this should interact in a more complex manner with the `match` below because starting to inhibit while some modifier is held (and in `suppressed_keys`) then releasing the modifier will currently leave it in `suppressed_keys`, which does not seem right.

This behavior should also be tested in the unit tests below.

Reply:
I've since updated the handling in a way that i think works with `supressed_keys` properly ([and i've commented the behaviour](https://github.com/YaLTeR/niri/pull/630/files#diff-306fabc3ade1a58278e40c0c99da8c0d45c5f2b7e7e43c3ead13d64099194aa2R2401-R2408)), but to be honest, thinking about the order of events kinda makes my head hurt and i'm dreading writing the requested tests. I'm also not entirely confident that my intuition of how it should behave exactly aligns with what you think it should do? Do you wanna take a shot at writing tests for this? :pleading_face::point_right::point_left:

Reply:
Seems fine to me. For the tests, just add a few cases with the shortcuts inhibited = true. For pressing and releasing allow-inhibit, !allow-inhibit, which also check suppress keys. Basically check that a shortcut that would return Intercept() returns instead Forward if inhibited = true. And maybe also something which first does a press with inhibited = true, then release with inhibited = false in case that logic would change later.